File: app/src/main/java/com/jakebarnby/filemanager/sources/onedrive/OneDriveFile.java
package com.jakebarnby.filemanager.sources.onedrive;

import com.jakebarnby.filemanager.sources.models.SourceFile;
import com.jakebarnby.filemanager.sources.models.SourceType;
import com.jakebarnby.filemanager.util.Constants;
import com.microsoft.graph.extensions.DriveItem;

/**
 * Created by Jake on 6/7/2017.
 */

public class OneDriveFile extends SourceFile {

    private String mDriveId;

    public OneDriveFile(DriveItem file) {
        setPath(file.webUrl);
        setDriveId(file.id);
        setName(file.name);
        setSourceType(SourceType.REMOTE);
        setSourceName(Constants.Sources.ONEDRIVE);
        setDirectory(file.folder != null);
        setSize(file.size);
        setModifiedTime(file.lastModifiedDateTime.getTimeInMillis());
        if (file.thumbnails != null && file.thumbnails.getCurrentPage().size() > 0) {
            setThumbnailLink(file.thumbnails.getCurrentPage().get(0).small.url);
        }
    }

    public String getDriveId() {
        return mDriveId;
    }

    public void setDriveId(String mDriveId) {
        this.mDriveId = mDriveId;
    }
}

File: app/src/main/java/com/jakebarnby/filemanager/sources/onedrive/OneDriveFile.kt
package com.jakebarnby.filemanager.sources.onedrive

import com.jakebarnby.filemanager.models.SourceFile
import com.jakebarnby.filemanager.models.SourceConnectionType
import com.jakebarnby.filemanager.models.SourceType
import com.jakebarnby.filemanager.util.Constants.Sources
import com.microsoft.graph.extensions.DriveItem

/**
 * Created by Jake on 6/7/2017.
 */
class OneDriveFile(file: DriveItem) : SourceFile() {

    var driveId: String

    init {
        path = file.webUrl
        driveId = file.id
        name = file.name
        sourceId = SourceType.ONEDRIVE.id
        isDirectory = file.folder != null
        size = file.size
        modifiedTime = file.lastModifiedDateTime.timeInMillis
        isHidden = file.name.startsWith(".")
        if (file.thumbnails != null && file.thumbnails.currentPage.size > 0) {
            thumbnailLink = file.thumbnails.currentPage[0].small.url
        }
    }
}